packages/rules/src/bannedDependencies.ts

```diff
@@ -73,24 +73,25 @@ function checkBanned(
     return;
   }
 
-  const expectedDependencies: Record<string, string> = {};
+  const violations: string[] = [];
 
   for (const dependency of Object.keys(dependencies)) {
     for (const bannedDependency of bannedDependencies) {
-      if (!minimatch(dependency, bannedDependency)) {
-        expectedDependencies[dependency] = dependencies[dependency];
+      if (minimatch(dependency, bannedDependency)) {
+        violations.push(dependency);
       }
     }
   }
 
-  if (Object.keys(expectedDependencies).length !== Object.keys(dependencies).length) {
+  if (violations.length > 0) {
+    const newPackageJson = { ...packageJson };
+    violations.forEach(v => delete newPackageJson[block]![v]);
```

Comment:
why doesn't  this happen as part of the for loop above?

Reply:
idk, i thought this was already a bit taboo pulling it out of the `fixer` scope.
do you think that's more readable?

Reply:
i think it's more readable, and we use it for the long message anyway?